mlflow/openai/__init__.py

```diff
@@ -698,91 +668,103 @@ def get_params_list(self, data):
-            max_tokens_per_minute=self.api_config.max_tokens_per_minute,
-        )
-        return [r["choices"][0]["message"]["content"] for r in results]
+
+        requests = [
+            partial(
+                self.client.with_options(
+                    max_retries=max_retries, timeout=timeout
+                ).chat.completions.create,
+                messages=message,
+                model=self.model["model"],
+                **params,
+            )
+            for message in messages_list
+        ]
+
+        results = process_api_requests(requests=requests)
+
+        return [r.choices[0].message.content for r in results]
 
     def _predict_completions(self, data, params):
         from mlflow.openai.api_request_parallel_processor import process_api_requests
 
         _validate_model_params(self.task, self.model, params)
         prompts_list = self.format_completions(self.get_params_list(data))
-
+        max_retries = params.pop("max_retries", self.api_config.max_retries)
+        timeout = params.pop("timeout", self.api_config.timeout)
         batch_size = params.pop("batch_size", self.api_config.batch_size)
         _logger.debug(f"Requests are being batched by {batch_size} samples.")
+
         requests = [
-            {
-                **self.model,
+            partial(
+                self.client.with_options(
+                    max_retries=max_retries, timeout=timeout
+                ).completions.create,
+                prompt=prompts_list[i : i + batch_size],
+                model=self.model["model"],
                 **params,
-                "prompt": prompts_list[i : i + batch_size],
-            }
+            )
             for i in range(0, len(prompts_list), batch_size)
         ]
-        request_url = self._construct_request_url("completions", REQUEST_URL_COMPLETIONS)
-
-        results = process_api_requests(
-            requests,
-            request_url,
-            api_token=self.api_token,
-            max_requests_per_minute=self.api_config.max_requests_per_minute,
-            max_tokens_per_minute=self.api_config.max_tokens_per_minute,
-        )
-        return [row["text"] for batch in results for row in batch["choices"]]
+
+        results = process_api_requests(requests=requests)
+
+        return [row.text for batch in results for row in batch.choices]
 
     def _predict_embeddings(self, data, params):
         from mlflow.openai.api_request_parallel_processor import process_api_requests
 
         _validate_model_params(self.task, self.model, params)
+        max_retries = params.pop("max_retries", self.api_config.max_retries)
+        timeout = params.pop("timeout", self.api_config.timeout)
         batch_size = params.pop("batch_size", self.api_config.batch_size)
         _logger.debug(f"Requests are being batched by {batch_size} samples.")
 
         first_string_column = _first_string_column(data)
         texts = data[first_string_column].tolist()
+
```

Comment:
```suggestion
        client = self.client.with_options(
            max_retries=max_retries, timeout=timeout
        )
```

can we update the options here?